bundles/org.eclipse.equinox.http.servlet/src/org/eclipse/equinox/http/servlet/internal/context/DispatchTargets.java

```diff
@@ -119,6 +119,20 @@ else if (dispatcherType == DispatcherType.FORWARD) {
 
 			requestWrapper.push(this);
 
+			if (dispatcherType == DispatcherType.INCLUDE) {
+				HttpServletResponse previous = null;
+				HttpServletResponse next = response;
+
+				while (next instanceof HttpServletResponseWrapper) {
+					previous = next;
+					next = (HttpServletResponse) ((HttpServletResponseWrapper) next).getResponse();
+				}
```

Comment:
Good question!

My goal was ensuring it works consistently for every ResponseWrappers in the chain. I think that simplification would allow bypassing the restriction by calling the corresponding methods on any ResponseWrappers behind IncludeDispatchResponseWrapper.

(Staying at my original test scenario in Liferay, HttpServletResponse is stored when initializing the PortletResponse object and accessed later through getHttpServletResponse(). Maybe that case would work (I'm not sure), but the possibility to manipulate other chain objects is still given.) 

We have to be careful to ensure consistent behaviour when any other object in the response chain is accessed for manipulation.

Do you agree?

(By the way, sorry for the late reaction, I really had to think it over. :) )